File: Source/Fantasy.Jobs/Resources/ResourceParameter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Runtime.Serialization;
using System.Reflection;

namespace Fantasy.Jobs.Resources
{
    [Serializable]
    public class ResourceParameter
    {
        public ResourceParameter(string name, object values = null)
        {
            if (name == null)
            {
                throw new ArgumentNullException("name");
            }
            this.Name = name;

           
            if (values != null)
            {
                if (values is IDictionary<string, string>)
                {
                    foreach (KeyValuePair<string, string> kv in (IDictionary<string, string>)values)
                    {
                        _values.Add(kv.Key, kv.Value);
                    }
                }
                else
                {
                    Type t = values.GetType();
                    foreach (PropertyInfo pi in t.GetProperties())
                    {
                        object v = pi.GetValue(values, null);
                        string s = v != null ? Convert.ToString(v) : null;
                        this._values.Add(pi.Name, s);
                    }
                }
                
            }
        }

        internal ResourceParameter()
        {

        }

        public string Name { get; internal set; }


        private Dictionary<string, string> _values = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);

        public Dictionary<string,string> Values
        {
            get { return _values; }
        }

        public override string ToString()
        {
            StringBuilder rs = new StringBuilder();
            rs.AppendFormat("name={0}", this.Name);
            foreach (KeyValuePair<string, string> item in this._values)
            {
                rs.AppendFormat(";{0}={1}", item.Key, item.Value);
            }

            return rs.ToString();

        }

    }
}

File: AutoConverted/Fantasy.Jobs/Resources/ResourceParameter.java
package Fantasy.Jobs.Resources;

public class ResourceParameter implements Serializable
{
//C# TO JAVA CONVERTER TODO TASK: C# optional parameters are not converted to Java:
//ORIGINAL LINE: public ResourceParameter(string name, object values = null)
	public ResourceParameter(String name, Object values)
	{
		if (name == null)
		{
			throw new ArgumentNullException("name");
		}
		this.setName(name);


		if (values != null)
		{
			if (values instanceof java.util.Map<String, String>)
			{
				for (java.util.Map.Entry<String, String> kv : (java.util.Map<String, String>)values)
				{
					_values.put(kv.getKey(), kv.getValue());
				}
			}
			else
			{
				java.lang.Class t = values.getClass();
				for (PropertyInfo pi : t.GetProperties())
				{
					Object v = pi.GetValue(values, null);
					String s = v != null ? String.valueOf(v) : null;
					this._values.put(pi.getName(), s);
				}
			}

		}
	}

	public ResourceParameter()
	{

	}

	private String privateName;
	public final String getName()
	{
		return privateName;
	}
	public final void setName(String value)
	{
		privateName = value;
	}


	private java.util.HashMap<String, String> _values = new java.util.HashMap<String, String>(StringComparer.OrdinalIgnoreCase);

	public final java.util.HashMap<String,String> getValues()
	{
		return _values;
	}

	@Override
	public String toString()
	{
		StringBuilder rs = new StringBuilder();
		rs.append(String.format("name=%1$s", this.getName()));
		for (java.util.Map.Entry<String, String> item : this._values.entrySet())
		{
			rs.append(String.format(";%1$s=%2$s", item.getKey(), item.getValue()));
		}

		return rs.toString();

	}

}